File: app/src/main/java/com/callmangement/EHR/support/SolidLine.java
package com.callmangement.EHR.support;

import com.itextpdf.text.pdf.PdfContentByte;

public class SolidLine implements LineDash {
    public void applyLineDash(PdfContentByte canvas) { }
}

class DottedLine implements LineDash {
    public void applyLineDash(PdfContentByte canvas) {
        canvas.setLineCap(PdfContentByte.LINE_CAP_ROUND);
        canvas.setLineDash(0, 4, 2);
    }
}

class DashedLine implements LineDash {
    public void applyLineDash(PdfContentByte canvas) {
        canvas.setLineDash(3, 3);
    }
}

File: app/src/main/java/com/callmangement/ehr/support/SolidLine.kt
package com.callmangement.ehr.support

import com.itextpdf.text.pdf.PdfContentByte

class SolidLine : LineDash {
    override fun applyLineDash(canvas: PdfContentByte?) {
        // Solid line: No dash
    }
}

class DottedLine : LineDash {
    override fun applyLineDash(canvas: PdfContentByte?) {
        canvas?.let {
            it.setLineCap(PdfContentByte.LINE_CAP_ROUND)
            it.setLineDash(0f, 4f, 2f)
        }
    }
}

class DashedLine : LineDash {
    override fun applyLineDash(canvas: PdfContentByte?) {
        canvas?.setLineDash(3f, 3f)
    }
}